aten/src/ATen/NamedTensorUtils.cpp

```diff
@@ -51,33 +51,38 @@ static void report_positional_error(
     const Dimname& name,
     const Dimname& other_name,
     DimnameList names,
-    DimnameList other_names) {
+    DimnameList other_names,
+    const char* action) {
   // TODO(zou3519): Can improve message by checking if names are alignable and suggesting workarounds
   TORCH_CHECK(false,
-      "Error when attempting to broadcast dims ", names, " and dims ",
+      "Error when attempting to ", action, " dims ", names, " and dims ",
```

Comment:
It might be worth truncating the string for `action` if someone has many names, though it might be rare.

Reply:
> It might be worth truncating the string for action if someone has many names, though it might be rare.

What do you mean by "truncating the string for action"?

Reply:
sorry I meant `names`